### PR TITLE
feat(parser): add source map line number tracking for precise error reporting

### PR DESCRIPTION
**Pull request by erraggy**
## Summary

Add "lossless" parsing capability inspired by [libopenapi](https://pb33f.io/libopenapi/about/) that preserves source line/column information from YAML/JSON documents. This enables precise error location reporting across all packages.

- ✅ New `SourceMap` type maps JSON paths to source locations (line:column:file)
- ✅ Opt-in via `WithSourceMap(true)` option (zero overhead when disabled)  
- ✅ CLI support: `--source-map` / `-s` flag on all commands
- ✅ IDE-friendly output format: `file:line:column: path: message`
- ✅ Dual-location tracking for joiner collisions (both conflicting sources)
- ✅ External file source maps built during `$ref` resolution
- ✅ `RefLocation` tracks both `$ref` origin and target locations

### CLI
```bash
# Validate with line numbers
oastools validate --source-map openapi.yaml
# Output: api.yaml:42:5: paths./users.get: missing required field 'responses'

# All commands support -s/--source-map
oastools convert -s -t 3.0.3 swagger.yaml
oastools diff -s api-v1.yaml api-v2.yaml
oastools fix -s openapi.yaml
```

### Library
```go
result, _ := parser.ParseWithOptions(
    parser.WithFilePath("openapi.yaml"),
    parser.WithSourceMap(true),
)

validateResult, _ := validator.ValidateWithOptions(
    validator.WithParsed(*result),
    validator.WithSourceMap(result.SourceMap),
)

for _, err := range validateResult.Errors {
    fmt.Printf("%s: %s\n", err.Location(), err.Message)
    // Output: openapi.yaml:42:5: missing required field 'responses'
}
```

## New Public API

**Types:**
- `parser.SourceLocation` - Position in source document (Line, Column, File)
- `parser.RefLocation` - Reference origin and target tracking
- `parser.SourceMap` - JSON path to location mapping

**Methods:**
- `parser.WithSourceMap(bool)` - Enable source map building
- `issues.Issue.Location()` - IDE-friendly location string
- `issues.Issue.HasLocation()` - Check if location is known
- `WithSourceMap` options on all packages

## Test plan

- [x] All 3,386 existing tests pass
- [x] New tests for SourceMap operations (nil safety, copy, merge)
- [x] Integration tests for each package
- [x] CLI flag verification
- [x] Edge cases: special characters in paths, external refs, multi-file joins

## Breaking Changes

**None** - All changes are additive and opt-in.

## Performance

- **Disabled (default):** Zero overhead
- **Enabled:** ~5-10% parsing overhead (proportional to document size)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: validator/validator.go [hunk 1]
```diff
@@ -1020,42 +1073,34 @@ func (v *Validator) validateOAS3Paths(doc *parser.OAS3Document, result *Validati
 			continue
 		}
 
+		pathPrefix := fmt.Sprintf("paths.%s", pathPattern)
```
[the remark this answers is not in the document]
Comment (erraggy) [reply validator/validator.go]: ℹ️ **No change needed** - After reviewing the code, `pathPrefix` is defined on line 1084 and used consistently throughout the function (lines 1088, 1096, 1107, 1122). The variable is properly scoped and all error messages in `validateOAS3Paths` use `pathPrefix` or derive from it. The line numbers may have shifted from the original review.

File: converter/converter.go [hunk 2]
```diff
@@ -460,12 +473,11 @@ func (c *Converter) convertOAS3ToOAS3(parseResult parser.ParseResult, targetVers
...
-		Path:     "openapi",
-		Message:  fmt.Sprintf("Updated version from %s to %s", parseResult.Version, result.TargetVersion),
-		Severity: SeverityInfo,
-		Context:  "OAS 3.x versions are generally compatible, but verify features are supported",
-	})
+	c.addIssueWithContext(result, "openapi", fmt.Sprintf("Updated version from %s to %s", parseResult.Version, result.TargetVersion), "OAS 3.x versions are generally compatible, but verify features are supported")
+	// Override severity to Info (addIssueWithContext defaults to Warning)
+	if len(result.Issues) > 0 {
+		result.Issues[len(result.Issues)-1].Severity = SeverityInfo
+	}
```
[the remark this answers is not in the document]
Comment (erraggy) [reply converter/converter.go]: ✅ **Fixed** - Added `addInfoWithContext` helper that defaults to `SeverityInfo` with context support. This eliminates the fragile pattern of modifying the last issue after appending.
